app/models/user_email.rb

```diff
@@ -22,6 +22,23 @@ class UserEmail < ActiveRecord::Base
   before_save -> { destroy_email_tokens(self.email_was) }, if: :will_save_change_to_email?
 
   after_destroy { destroy_email_tokens(self.email) }
+  def self.ensure_consistency!
+    user_ids_without_primary_email = DB.query_single <<~SQL
+      SELECT u.id
+      FROM users u
+      LEFT JOIN user_emails ue ON u.id = ue.user_id AND ue.primary = true
+      WHERE ue.id IS NULL;
+    SQL
+
+    user_ids_without_primary_email.each do |user_id|
+      UserEmail.create!(
```

Comment:
Using `create` instead of a batch operation here to ensure we don't miss any hooks/validations - this performance cost should be ok since we don't expect a ton of such missing primary user emails (it's not something that is expected to occur in the normal course of operations for our application).